### PR TITLE
Add basic `/organizations` endpoints

### PR DESCRIPTION
**Pull request by slifty**
This PR adds a new `Organization` entity type along with the basic create / read endpoints for it.

This PR does *not* establish relationships between organizations and proposals, or organizations and non-proposal data.  It also does not yet create a public / non-authenticated user view of organization data.

Related to #777

### CODE REVIEW

File: src/openapi.json
Comment (reefdog) [inline src/openapi.json:152]: I think this list is inaccurate? Probably copypasta from Applicant!

File: src/types/Organization.ts
Comment (reefdog) [inline src/types/Organization.ts:28]: Just for my own edification, why is this an array?

File: src/__tests__/organizations.int.test.ts
Comment (reefdog) [inline src/__tests__/organizations.int.test.ts:16]: While we don't have format validation, it appears this is meant to mimic a real EIN, which means it should be nine digits. (I'm pretty confident you intended this, and e.g. `11-1111111` conforms properly.)

Only flagging it once but I see it several times:

```suggestion
		employerIdentificationNumber: '22-2222222',
```

File: src/__tests__/organizations.int.test.ts
Comment (reefdog) [inline src/__tests__/organizations.int.test.ts:148]: Only flagging this once, but I see it several times. (But not always! There are correct nine-digit versions as well.)

```suggestion
					employerIdentificationNumber: '11-1111111',
```

File: src/openapi.json
Comment (reefdog) [inline src/openapi.json:140]: Should be consistent with other examples (and expected EIN format).

```suggestion
						"example": "11-1111111"
```

**Review by reefdog — COMMENTED** [review]
For the most part, this is great! I found an errant bit of copypasta and some malformed sample EINs. The only significant question I have is why there isn't a POST example in the Swagger spec? Was this an oversight? It stymied me!

**Review by reefdog — CHANGES_REQUESTED** [review]
Bah, I meant to request changes with my review just then, but clicked the wrong radio button. You don't get off that easy!

**Comment by slifty** [comment]
@reefdog thanks for the review!  FWIW, and maybe this is an anti-pattern, but I'll often mark a PR as approved with some requested changes in the event that the requested changes are minor enough that I don't think I'll need to re-review (e.g. fixing a typo etc).

File: src/openapi.json
Comment (slifty) [reply src/openapi.json:152]: Phewf thanks -- a little disappointed the linter didn't catch that!

**Comment by reefdog** [comment]
> FWIW, and maybe this is an anti-pattern, but I'll often mark a PR as approved with some requested changes in the event that the requested changes are minor enough that I don't think I'll need to re-review (e.g. fixing a typo etc).

I've noticed that, and I **do** think that's an anti-pattern, and that you should be comfortable leaving a "request changes" review. It's not a failing grade, it's a communication signal: "In my opinion as reviewer, the PR shouldn't merge to `main` until some changes are made." I reserve "approve" for when there are no changes requested, or when they are subjective enough that I don't mind the dev discarding them.

I do wish we could signal "if you make these necessary changes, approval will be automatic and re-review isn't necessary" but we can state that in the comment. (I do grok that there's still a cost here, as the reviewer now has to come back and re-review, but I think that's an acceptable cost compared to losing the clarity of approve v. request changes).

Have I convinced you¹??

¹ that you not longer want to work with me and regret ever introducing yourself at SRCCON

**Review by reefdog — CHANGES_REQUESTED** [review]
I have one minimal request — looks like your EIN sample fix got fixed TOO hard! — but also still don't see a POST example in the generated Swagger spec. Shouldn't there be one?

Organization Swagger spec

**Comment by reefdog** [comment]
Oh my goodness @slifty I just realized you hadn't yet requested a re-review from me. 🤦 I'm sorry for jumping the gun with the above and hope that wasn't frustrating! I'll hold tight until you're ready.

If it helps I'm just really excited about this functionality.

File: src/__tests__/organizations.int.test.ts
Comment (slifty) [reply src/__tests__/organizations.int.test.ts:12]: bah -- tbh I wonder if it might be better to have data that clearly does not even try to conform, since we're doing no checks whatsoever yet.

**Comment by slifty** [comment]
> I do wish we could signal "if you make these necessary changes, approval will be automatic and re-review isn't necessary" but we can state that in the comment

That's *sometimes* the intention (and to your point someone can press the bright red "merge without formal approval" button I suppose).

That said sometimes the feedback really is meant as "take this or leave it; it's approved as is, but these changes might be worth considering"

Anyway I hear ya and will try to take your advice of not working with you any more.

File: src/__tests__/organizations.int.test.ts
Comment (reefdog) [reply src/__tests__/organizations.int.test.ts]: Thanks! If we want to demonstrate/test alternative tax IDs, that's cool, we should just call that out explicitly. These are pretty clearly meant to conform to US EIN format — so I appreciate taking the trouble. 🙏

**Comment by reefdog** [comment]
> That said sometimes the feedback really is meant as "take this or leave it; it's approved as is, but these changes might be worth considering"

Totally, in which case I think "Approved" is the right call. The one pattern I'm trying to avoid baking in is using "Approved" when you _do_ have very real and necessary changes that you are requesting, but then to signal re-review isn't necessary after resolution.

> Anyway I hear ya and will try to take your advice of not working with you any more.

You can't take this back just because you said it on a day that doesn't exist. (Checks timestamp.) Darnit, I mean, I read it on a day that doesn't exist.

File: CHANGELOG.md
Comment (reefdog) [inline CHANGELOG.md:8]: Leap day release!

whoooa

File: CHANGELOG.md
Comment (reefdog) [inline CHANGELOG.md:13]: Whoops just noticed this typo.

```suggestion
- The GET `/organizations` endpoint now exists.
```

**Review by reefdog — CHANGES_REQUESTED** [review]
Ironically, given our discussion, I now have to follow through on "request changes" for literally just a typo fix. 😞 This is boding well for never working together again.

That said, I *also* noticed that the Swagger spec now includes a POST example, but… it's under the heading "Organization" instead of "Organizations", while all the other entities bundle GET and POST requests under a single entity header? Can that be fixed?

Organizations Swagger

**Comment by slifty** [comment]
@reefdog hey look ok, you're the one who has to re-review!

Fixed both typos!

**Review by reefdog — APPROVED** [review]
WHEEEEEE

**Comment by reefdog** [comment]
@slifty I hit approve, but… shouldn't you actually increment the version in `package.json` (not just changelog and openapi)?

**Comment by slifty** [comment]
@reefdog since the last time we had a version we decided that we're versioning the API spec -- the package version is something of a misnomer and isn't used by anything.  I'd say we should either leave it as is, or open an issue to just set it to 0.0.0 or something that is clearly-not-a-thing.

**Comment by reefdog** [comment]
@slifty That makes sense, thanks for the explanation! No problem, merge as-is and let's open an issue to zero it out (or whatever).

**Pull request merged**
